backend/src/main/java/gov/cdc/usds/simplereport/service/ResultsUploaderDeviceService.java

```diff
@@ -0,0 +1,100 @@
+package gov.cdc.usds.simplereport.service;
+
+import gov.cdc.usds.simplereport.config.FeatureFlagsConfig;
+import gov.cdc.usds.simplereport.db.model.DeviceType;
+import gov.cdc.usds.simplereport.db.model.DeviceTypeDisease;
+import gov.cdc.usds.simplereport.db.model.SupportedDisease;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import org.springframework.stereotype.Component;
+
+// Util class to extend ResultsUploaderCachingService that
+// take advantage of the @Cacheable methods in that class, since @Cacheable's won't
+// be used if called in the same class, as well as to centralize some checking logic
+// for the cached device map.
+@Component
+public class ResultsUploaderDeviceService {
+  private final ResultsUploaderCachingService resultsUploaderCachingService;
+  private final FeatureFlagsConfig featureFlagsConfig;
+
+  public ResultsUploaderDeviceService(
+      ResultsUploaderCachingService resultsUploaderCachingService,
+      FeatureFlagsConfig featureFlagsConfig) {
+    this.resultsUploaderCachingService = resultsUploaderCachingService;
+    this.featureFlagsConfig = featureFlagsConfig;
+  }
+
+  private static String removeTrailingAsterisk(String value) {
```

Comment:
actually looking at the LIVD API again / the device names, it looks like the model names don't actually have the asterisks anymore / might have gotten cleaned up after we switched over from the spreadhseet. will just go ahead and delete this code since I think it was made obsolete